### ISSUE TITLE
Problem when using MenuFilter or RefinementListFilter on boolean properties.

### ISSUE THREAD

**Issue by richtera** [searchkit/searchkit#525]
It works mostly, but when refreshing the screen it will not find the key according to the URL since the internal keys are integer but the keys from the query are strings. ES has added a key_as_string property which might be useful here and I see a similar issue mentioning using an integer value for filtering which would have similar problems.

**Comment by ssetem** [comment]
Yeh issue with serialize/deserialize.

Do you mind pasting an example of the url which includes the integer keys?

**Comment by richtera** [comment]
Unfortunately I don't have a public site showing this right now. I will try to do a fork.

**Comment by richtera** [comment]
I see a similar issue using
```jsx
<RefinementListFilter operator="OR" orderKey="_term" orderDirection="desc" field="stats.year" id="years" title="Year" fieldOptions={{type: "nested", options: {path: "stats"}}} size={20}/>
```
Selecting some items (the field is nested and of type long.
I get the URL as:
```
...ate?years[0]=2008&years[1]=2009
```
The list looks like this:
![image](https://user-images.githubusercontent.com/708186/31255760-82d0a022-a9fc-11e7-9509-0d5522a12520.png)

But refreshing keeps the URL and "the filter" but not the List checkboxes.
![image](https://user-images.githubusercontent.com/708186/31255777-a7bc5868-a9fc-11e7-8cc7-02900ff35567.png)

But the list itself shows up without checkboxes. NOTE: That initially I can see the screen flash with the correct image of the checkboxes but then when the display settles they get unchecked and re-render.

![image](https://user-images.githubusercontent.com/708186/31255791-b6b2e5f8-a9fc-11e7-91d4-efa0509e9659.png)

Then checking an item... the URL changes:
```
...te?years[0]=2008&years[1]=2009&years[2]=2008
```

But the current version no longer causes an error in the DSL so that part seems ok (at least during this test)

**Comment by ssetem** [comment]
thanks, I see the issue now

**Comment by ssetem** [comment]
@richtera 
this should be fixed in beta version 2.3.0-0

thanks!

**Comment by richtera** [comment]
Works in 2.3.0-0
